pkg/getty/etc/profile.d/namespace.sh

```diff
@@ -1 +1,23 @@
-export PS1="(ns: getty) $PS1"
+#/bin/sh
+
+in_a_ns() {
+	IN_A_NS=0
+	NOT_IN_A_NS=1
+
+	# If PID 1 != /sbin/init then we are in a container
+	if ! pgrep /sbin/init > /dev/null; then
+		return $IN_A_NS
+	fi
+
+	# If containerd or runc are in the pstree then we're probably in a namespace
+	if pstree | grep -qE "(containerd|runc)"; then
+		return $IN_A_NS
+	fi
```

Comment:
> when you have an init getty, you don't see containerd or runc

This is because `/etc/init.d/001-getty` is calling `/usr/bin/rungetty.sh` synchronously and blocking the rest of the boot process -- i.e. containerd hasn't been started yet. This is (of course) a bug.